From 6e4081f1be93fc0afa91c6c0f02b4f269c24af3a Mon Sep 17 00:00:00 2001
From: "Vaidya, Mahesh" <mahesh.vaidya@altera.com>
Date: Tue, 4 Nov 2025 01:31:30 -0800
Subject: [PATCH] DTS: Add node to support pcie legacy interrupts

Add legacy-interrupt-controller node to support pcie legacy interrupt.

Upstream-Status: Pending
Signed-off-by: Vaidya, Mahesh <mahesh.vaidya@altera.com>
---
 .../dts/intel/socfpga_agilex_pcie_root_port.dtsi   | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/socfpga_agilex_pcie_root_port.dtsi b/arch/arm64/boot/dts/intel/socfpga_agilex_pcie_root_port.dtsi
index d8ed797d08bb..6e655b147531 100644
--- a/arch/arm64/boot/dts/intel/socfpga_agilex_pcie_root_port.dtsi
+++ b/arch/arm64/boot/dts/intel/socfpga_agilex_pcie_root_port.dtsi
@@ -24,7 +24,6 @@ pcie_0_pcie_aglx: pcie@200000000 {
 			reg-names = "Txs", "Cra", "Hip";
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_SPI 0x14 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-controller;
 			#interrupt-cells = <0x1>;
 			device_type = "pci";
 			bus-range = <0x0000000 0x000000ff>;
@@ -33,11 +32,16 @@ pcie_0_pcie_aglx: pcie@200000000 {
 			#address-cells = <0x3>;
 			#size-cells = <0x2>;
 			interrupt-map-mask = <0x0 0x0 0x0 0x7>;
-			interrupt-map = <0x0 0x0 0x0 0x1 &pcie_0_pcie_aglx 0x1>,
-					<0x0 0x0 0x0 0x2 &pcie_0_pcie_aglx 0x2>,
-					<0x0 0x0 0x0 0x3 &pcie_0_pcie_aglx 0x3>,
-					<0x0 0x0 0x0 0x4 &pcie_0_pcie_aglx 0x4>;
+			interrupt-map = <0x0 0x0 0x0 0x1 &pcie_intc 0x1>,
+					<0x0 0x0 0x0 0x2 &pcie_intc 0x2>,
+					<0x0 0x0 0x0 0x3 &pcie_intc 0x3>,
+					<0x0 0x0 0x0 0x4 &pcie_intc 0x4>;
 			status = "disabled";
+			pcie_intc: legacy-interrupt-controller {
+				interrupt-controller;
+				#address-cells = <0>;
+				#interrupt-cells = <1>;
+			};
 		};
 
 		pcie_0_msi_irq: msi@10008080 {
-- 
2.34.1

